zipline/data/us_equity_minutes.py

```diff
@@ -0,0 +1,316 @@
+            vol_col = np.zeros(minutes_count, dtype=np.uint32)
+
+            for row in df.iterrows():
+                dt = row[0]
+                idx = minutes.searchsorted(dt)
+
+                dt_col[idx] = dt.value / 1e9
+                open_col[idx] = row[1].loc["open"]
+                high_col[idx] = row[1].loc["high"]
+                low_col[idx] = row[1].loc["low"]
+                close_col[idx] = row[1].loc["close"]
+                vol_col[idx] = row[1].loc["volume"]
+
+            ctable(
+                columns=[
+                    open_col,
+                    high_col,
+                    low_col,
+                    close_col,
+                    vol_col,
+                    dt_col
+                ],
+                names=[
+                    "open",
+                    "high",
+                    "low",
+                    "close",
+                    "volume",
+                    "dt"
+                ],
+                rootdir=path,
+                mode='w'
+            )
+
+
+class MinuteBarWriterFromDataFrames(BcolzMinuteBarWriter):
+    _csv_dtypes = {
+        'open': float64,
+        'high': float64,
+        'low': float64,
+        'close': float64,
+        'volume': float64,
+    }
+
+    def __init__(self, first_trading_day):
+        self._first_trading_day = first_trading_day
+
+    def gen_frames(self, assets):
+        for asset in assets:
+            df = assets[asset]
+            yield asset, df.set_index("minute")
+
+
+class MinuteBarWriterFromCSVs(BcolzMinuteBarWriter):
+    """
+    BcolzMinuteBarWriter constructed from a map of CSVs to assets.
+
+    Parameters
+    ----------
+    asset_map: dict
+        A map from asset_id -> path to csv with data for that asset.
+
+    CSVs should have the following columns:
+        minute : datetime64
+        open : float64
+        high : float64
+        low : float64
+        close : float64
+        volume : int64
+    """
+    _csv_dtypes = {
+        'open': float64,
+        'high': float64,
+        'low': float64,
+        'close': float64,
+        'volume': float64,
+    }
+
+    def __init__(self, asset_map, first_trading_day):
+        self._asset_map = asset_map
+        self._first_trading_day = first_trading_day
+
+    def gen_frames(self, assets):
+        """
+        Read CSVs as DataFrames from our asset map.
+        """
+        dtypes = self._csv_dtypes
+
+        for asset in assets:
+            path = self._asset_map.get(asset)
+            if path is None:
+                raise KeyError("No path supplied for asset %s" % asset)
+            df = read_csv(path, parse_dates=['minute'], dtype=dtypes)
+            df = df.set_index("minute").tz_localize("UTC")
+
+            yield asset, df
+
+
+class BcolzMinuteBarReader(object):
+
+    def __init__(self, rootdir, sid_path_func=None):
+        self.rootdir = rootdir
+
+        metadata = self._get_metadata()
```

Comment:
The `write_metadata` counterpart to this is a free function at module scope, but this is an instancemethod.  I'd suggest moving this to module scope as well and just passing `self.rootdir`.